jetty-websocket/websocket-core/src/test/java/org/eclipse/jetty/websocket/core/autobahn/AutobahnWebSocketClient.java

```diff
@@ -132,121 +109,110 @@ public static void main(String[] args)
+            LOG.warn("Test Failed", t);
         }
         finally
         {
             if (client != null)
-            {
                 client.shutdown();
-            }
         }
-        System.exit(0);
     }
 
-    private Logger log;
+    private static final Logger LOG = Log.getLogger(AutobahnWebSocketClient.class);
     private URI baseWebsocketUri;
     private WebSocketCoreClient client;
-    private String hostname;
-    private int port;
     private String userAgent;
 
     public AutobahnWebSocketClient(String hostname, int port, String userAgent) throws Exception
     {
-        this.log = Log.getLogger(this.getClass());
-        this.hostname = hostname;
-        this.port = port;
         this.userAgent = userAgent;
         this.baseWebsocketUri = new URI("ws://" + hostname + ":" + port);
         this.client = new WebSocketCoreClient();
-
-        // TODO: this should be enabled by default
-        // this.client.getExtensionFactory().register("permessage-deflate",PerMessageDeflateExtension.class);
         this.client.start();
     }
 
-    public int getCaseCount() throws IOException, InterruptedException, ExecutionException, TimeoutException
+    public int getCaseCount() throws IOException, InterruptedException
     {
         URI wsUri = baseWebsocketUri.resolve("/getCaseCount");
-        GetCaseCountHandler onCaseCount = new GetCaseCountHandler();
+        TestMessageHandler onCaseCount = new TestMessageHandler();
         Future<FrameHandler.CoreSession> response = client.connect(onCaseCount, wsUri);
 
         if (waitForUpgrade(wsUri, response))
         {
-            onCaseCount.awaitMessage();
-            if (onCaseCount.hasCaseCount())
-            {
-                return onCaseCount.getCaseCount();
-            }
+            String msg = onCaseCount.textMessages.poll(2, TimeUnit.SECONDS);
+            onCaseCount.getCoreSession().abort(); // Don't expect normal close
+            assertTrue(onCaseCount.closeLatch.await(2, TimeUnit.SECONDS));
+            assertNotNull(msg);
+            return Integer.decode(msg);
         }
         throw new IllegalStateException("Unable to get Case Count");
     }
 
     public void runCaseByNumber(int caseNumber) throws IOException, InterruptedException
     {
         URI wsUri = baseWebsocketUri.resolve("/runCase?case=" + caseNumber + "&agent=" + UrlEncoded.encodeString(userAgent));
-        log.info("test uri: {}", wsUri);
-        EchoHandler onEchoMessage = new EchoHandler(caseNumber);
-
-        Future<FrameHandler.CoreSession> response = client.connect(onEchoMessage, wsUri);
+        LOG.info("test uri: {}", wsUri);
 
+        AutobahnFrameHandler echoHandler = new AutobahnFrameHandler();
+        Future<FrameHandler.CoreSession> response = client.connect(echoHandler, wsUri);
         if (waitForUpgrade(wsUri, response))
         {
-            onEchoMessage.awaitClose();
+            if (!echoHandler.closeLatch.await(5, TimeUnit.MINUTES))
```

Comment:
And 5 minutes is also a strange time? have a least a comment describing why

Reply:
some of the tests allow are quite long and say they want the response in under 1000 seconds (about 16 min), but 5 minutes seems easily long enough for us to respond to those cases